src/cpu/SearchBlock.zig

```diff
@@ -0,0 +1,344 @@
+pub fn calculateSliminess(
+    self: *@This(),
+    params: slimy.SearchParams,
+    context: anytype,
+    comptime resultCallback: fn (@TypeOf(context), slimy.Result) void,
+) usize {
+    var sufficiently_slimy_chunks: usize = 0;
+    for (0..size - mask.len + 1) |x| {
+        for (0..size - mask.len + 1) |z| {
+            var count: u8 = 0;
+
+            var preprocessed_count_1: u16 = 0;
+            inline for (use_preprocessed[0..15]) |location| {
+                preprocessed_count_1 += @as(u8, @bitCast(self.data[x * size + z + location.x * size + location.z]));
+            }
+            var preprocessed_count_2: u16 = 0;
+            inline for (use_preprocessed[15..]) |location| {
+                preprocessed_count_2 += @as(u8, @bitCast(self.data[x * size + z + location.x * size + location.z]));
+            }
+            count += @intCast((preprocessed_count_1 >> 4) + (preprocessed_count_2 >> 4));
+
+            var add_count: u16 = 0;
+            inline for (add) |location| add_count += @as(u8, @bitCast(self.data[x * size + z + location.x * size + location.z]));
+            count += @intCast(add_count & 0xf);
+
+            var sub_count: u16 = 0;
+            inline for (sub) |location| sub_count += @as(u8, @bitCast(self.data[x * size + z + location.x * size + location.z]));
+            count -= @intCast(sub_count & 0xf);
+
+            if (count >= params.threshold) {
+                @setCold(true);
+                sufficiently_slimy_chunks += 1;
+                const real_x = @as(i32, @intCast(x + offset)) + self.min_x;
+                const real_z = @as(i32, @intCast(z + offset)) + self.min_z;
+
+                if (real_x >= params.x0 and real_x < params.x1 and
+                    real_z >= params.z0 and real_z < params.z1)
+                {
+                    resultCallback(context, .{
+                        .x = real_x,
+                        .z = real_z,
+                        .count = count,
+                    });
+                }
+            }
+        }
+    }
+    return sufficiently_slimy_chunks;
+}
+
+/// For every chunk within the searched area defined by this `SearchBlock`
+/// checks whether the amount of slime chunk slime chunks in spawn range of a player
+/// at the center of each chunk meets the given `threshold`
+pub fn calculateSliminessUncached(
+    self: *@This(),
+    comptime threshold: u8,
+    context: anytype,
+    comptime resultCallback: fn (@TypeOf(context), slimy.Result) void,
+) void {
+    for (0..size - mask.len) |x| {
+        for (0..size - mask.len) |z| {
+            var count: u8 = 0;
+            for (comptime getDistMask(), 0..) |row, i| {
+                for (row, 0..) |b, j| {
+                    count += @intFromBool(b and @import("slime_check/scalar.zig").isSlime(0x51133, @intCast(x + i), @intCast(z + j)));
+                }
+            }
+            if (count >= threshold) {
+                @setCold(true);
+                // @branchHint(.unlikely);
```

Comment:
if it's reasonable to test this on a nightly build, i would consider updating if `@branchHint` has a noticeable impact here. If it doesn't, just remove it :)